### PR TITLE
Making datastore Connection.commit() return low-level protobuf.

### PR DESCRIPTION
**Pull request by dhermes**
Towards #2746. This approach is to slowly transition from our current approach to use the GAPIC generated surface.

It is unfortunately tangled quite a bit (partly because we may have too much mocked in the tests).

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes so this is preparatory to adding `google.cloud.datastore._gax`, with its own implemenation of the API via the GAPIC client?

**Comment by lukesneeringer** [comment]
@dhermes Is your goal to get this in with the umbrella releaase, or after?

File: datastore/google/cloud/datastore/batch.py [hunk 1]
```diff
@@ -311,3 +312,21 @@ def _assign_entity_to_pb(entity_pb, entity):
...
+
+    :type commit_response_pb: :class:`.datastore_pb2.CommitResponse`
+    :param commit_response_pb: The protobuf response from a commit request.
+
+    :rtype: tuple
+    :returns: The pair of the number of index updates and a list of
+              :class:`.entity_pb2.Key` for each incomplete key
+              that was completed in the commit.
+    """
+    mut_results = commit_response_pb.mutation_results
```
Comment (daspecster) [inline datastore/google/cloud/datastore/batch.py]: Could just be me, but `mut_results` feels like `MUT` from the unite tests.

File: datastore/unit_tests/test_batch.py [hunk 2]
```diff
@@ -369,30 +369,62 @@ def begin(self):
...
+                    entity_pb2.Key.PathElement(
+                        kind='Bar',
+                        name='baz',
+                    ),
+                ],
+            ),
+        ]
+        response = datastore_pb2.CommitResponse(
+            mutation_results=[
+                datastore_pb2.MutationResult(key=key) for key in keys
```
Comment (daspecster) [inline datastore/unit_tests/test_batch.py]: I'm surprised key/keys aren't reserved words, but it doesn't look like they are.
https://docs.python.org/2.5/ref/keywords.html

Comment (lukesneeringer) [inline datastore/unit_tests/test_batch.py]: They are not reserved words, and `key` is an extremely common variable name.

File: datastore/google/cloud/datastore/batch.py (hunk 1 above)
Comment (lukesneeringer) [inline datastore/google/cloud/datastore/batch.py]: It is a four line function. It is fine.

**Comment by lukesneeringer** [comment]
Approved, but holding off on merge until the umbrella release question is answered.

File: datastore/google/cloud/datastore/batch.py (hunk 1 above)
Comment (dhermes) [inline datastore/google/cloud/datastore/batch.py]: Also the `mutation_results` on the RHS of the line should make it clear

**Comment by dhermes** [comment]
@lukesneeringer A merge is just fine.

@tseaver `google.cloud.datastore._gax` already exists, but yes this is part of the transition to switch to using the GAPIC layer in `google.cloud.datastore._gax`.

**Comment by dhermes** [comment]
Passing on Travis, not done yet on CircleCI (since it doesn't **yet** do a restricted set of tests). Merging before CircleCI is done.

**Pull request merged**